elephant/signal_processing.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+"""
+docstring goes here.
+
+:copyright: Copyright 2015 by the Elephant team, see AUTHORS.txt.
+:license: Modified BSD, see LICENSE.txt for details.
+"""
+
+import numpy as np
+import scipy.signal
+import quantities as pq
+import neo
+
+
+def butter(anasig, highpassfreq=None, lowpassfreq=None, order=4,
+           filtfunc='filtfilt'):
+    """
+    Butterworth filtering function for AnalogSignal objects. Filter type is
+    determined according to how values of highpassfreq and lowpassfreq are
+    given (see Parameters for details).
+
+    Parameters
+    ----------
+    anasig: AnalogSignal
```

Comment:
Also, as noted for the other open pull requests, the function should operate on `AnalogSignalArray`.